File: src/tools/community-goals.ts
import { z } from "zod";
import { type ToolRegistrar, jsonResult } from "./context.js";

export const registerCommunityGoalTools: ToolRegistrar = (server, ctx) => {
  server.tool(
    "list_community_goals",
    "List all community goals. Returns id, name, description, target amount, " +
      "current progress, status, repeatability, and times achieved.",
    {},
    async () => {
      const data = await ctx.client.listCommunityGoals();
      return jsonResult(data);
    },
  );

  server.tool(
    "get_community_goal",
    "Get full details of a single community goal by its ID.",
    {
      goal_id: z.number().int().describe("Community goal ID"),
    },
    async ({ goal_id }) => {
      const data = await ctx.client.getCommunityGoal(goal_id);
      return jsonResult(data);
    },
  );
};

File: src/tebex_mcp/tools/community_goals.py
"""Community goal tools."""

from __future__ import annotations

from typing import Annotated, Any

from fastmcp import FastMCP
from pydantic import Field

from tebex_mcp.context import ToolContext
from tebex_mcp.tools._common import map_tebex_errors


def register(mcp: FastMCP, ctx: ToolContext) -> None:
    @mcp.tool(
        name="list_community_goals",
        description="List all community goals with target, progress, status, and times achieved.",
    )
    @map_tebex_errors
    async def list_community_goals() -> Any:
        return await ctx.client.list_community_goals()

    @mcp.tool(
        name="get_community_goal",
        description="Get community goal details by ID.",
    )
    @map_tebex_errors
    async def get_community_goal(
        goal_id: Annotated[int, Field(description="Community goal ID", ge=1)],
    ) -> Any:
        return await ctx.client.get_community_goal(goal_id)
